.gitignore

```diff
@@ -1,17 +1,83 @@
-*.iml
-*.ipr
+
+# Created by https://www.gitignore.io/api/java,jetbrains,gradle
```

Comment:
Maybe... but that's how all of the class files that weren't output to the `build/` folder ended up being pushed to the repo.  It seemed more efficient and robust to me to just generate a gitignore file rather than go through the content by hand. I'll add the eclipse files.